front/src/pages/Index.vue

```diff
@@ -1,97 +1,141 @@
+import { whenever } from '@vueuse/core';
+import CardList from 'src/components/CTF/CardList.vue';
 import ctfnote from 'src/ctfnote';
-import { defineComponent } from 'vue';
-
-export default defineComponent({
-  name: 'PageIndex',
-  components: {},
-  setup() {
-    return {
-      me: ctfnote.me.injectMe(),
-      tabs: [
-        {
-          label: 'Incoming',
-          icon: 'query_builder',
-          route: { name: 'ctfs-incoming' },
-        },
-        {
-          label: 'Past',
-          icon: 'archive',
-          route: { name: 'ctfs-past' },
-        },
-        {
-          label: 'calendar',
-          icon: 'calendar_today',
-          route: { name: 'ctfs-calendar' },
-        },
-      ],
-    };
-  },
-  methods: {
-    openCreateCtfDialog() {
-      this.$q.dialog({
-        component: EditCtfDialogVue,
-      });
-    },
-    openImportCtfDialog() {
-      this.$q.dialog({
-        component: ImportCtfDialogVue,
-      });
-    },
-  },
+import { ShortDate, shortDate } from 'src/utils/shortDate';
+import { computed, nextTick, onMounted, ref, toRaw } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
+import CtfCalendar from '../components/CTF/CtfCalendar.vue';
+
+const router = useRouter();
+const route = useRoute();
+
+const currentDate = new Date();
+
+const params = computed(() => {
+  const year =
+    parseInt(route.params.year as string, 10) || currentDate.getFullYear();
+  const month = Math.min(
+    12,
+    Math.max(
+      1,
+      parseInt(route.params.month as string, 10) || currentDate.getMonth() + 1
+    )
+  );
+  return { year, month };
 });
+
+function scrollToCtf(ctfId: number, behavior: ScrollBehavior = 'smooth') {
+  document.querySelector(`[data-ctf-id="${ctfId}"]`)?.scrollIntoView({
+    behavior,
+    block: 'start',
+  });
+}
+
+const start = ref(toRaw(params.value));
+const end = ref(toRaw(params.value));
+
+const { ctfs, loaded, fetchMore } = ctfnote.ctfs.useCtfsByDate();
+
+function showNextCtf() {
+  for (const ctf of ctfs.value) {
+    if (ctf.endTime > currentDate) {
+      scrollToCtf(ctf.id, 'smooth');
+      break;
+    }
+  }
+}
+
+async function showToday() {
+  const date = shortDate.fromDate(new Date());
+  await doFetchMore(date);
+  await nextTick(showNextCtf);
+}
+
+// When the page is loaded for the first time, scroll to the next CTF
+whenever(() => ctfs.value.length, showNextCtf, { flush: 'post', once: true });
+
+async function handleNavigation(date: ShortDate) {
+  void router.replace({
+    name: 'ctfs',
+    params: { year: date.year, month: date.month },
+  });
+  await doFetchMore(date);
+  void nextTick(() => {
+    document
+      .querySelector(`[data-date="${shortDate.toId(date)}"]`)
+      ?.scrollIntoView({
+        behavior: 'smooth',
+        block: 'start',
+      });
+  });
+  if (shortDate.lt(date, start.value)) {
+    start.value = shortDate.copy(date);
+  }
+  if (shortDate.gt(date, end.value)) {
+    end.value = shortDate.copy(date);
+  }
+}
```

Comment:
I have this annoying bug that will not trigger by a refresh but does trigger when navigating back to the front page (video explains more than words):
[ctfnote bug scrolling.webm](https://github.com/user-attachments/assets/88165f2f-0ede-48d3-bd1a-cb6fc784978a)